src/main/java/com/compstore/config/ProdSecurityConfig.java

```diff
@@ -0,0 +1,39 @@
+package com.compstore.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity
+@RequiredArgsConstructor
+@Profile("!dev-h2")
+public class ProdSecurityConfig {
+
+    private final JwtAuthConverter jwtAuthConverter;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http.csrf(CsrfConfigurer::disable)
+                .authorizeHttpRequests(customizer -> customizer.anyRequest().authenticated())
+                .oauth2ResourceServer(
+                        httpSecurityOAuth2ResourceServerConfigurer ->
+                                httpSecurityOAuth2ResourceServerConfigurer.jwt(
+                                        jwtConfigurer ->
+                                                jwtConfigurer.jwtAuthenticationConverter(
```

Comment:
Wymagane użycie konwertera, gdyż keycloak ma zapisane role jako `admin`, a spring boot zapisuje role z przedrostkiem `ROLE_`, więc trzeba to przemapować